redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1286 @@
+                    _temp = copy(self._server_mutes[guild.id][user.id])
+                    del self._server_mutes[guild.id][user.id]
+            try:
+                await user.remove_roles(role, reason=reason)
+            except discord.errors.Forbidden:
+                if _temp:
+                    self._server_mutes[guild.id][user.id] = _temp
+                return False, _(MUTE_UNMUTE_ISSUES["permissions_issue"])
+            return True, None
+        else:
+            mute_success = []
+            tasks = []
+            for channel in guild.channels:
+                tasks.append(self.channel_unmute_user(guild, channel, author, user, reason))
+            results = await asyncio.gather(*tasks)
+            for task in results:
+                if not task["success"]:
+                    chan = task["channel"].mention
+                    issue = task["reason"]
+                    mute_success.append(f"{chan} - {issue}")
+            await self.config.member(user).clear()
+            if mute_success:
+                return False, "\n".join(s for s in mute_success)
+            else:
+                return True, None
+
+    async def channel_mute_user(
+        self,
+        guild: discord.Guild,
+        channel: discord.abc.GuildChannel,
+        author: discord.Member,
+        user: discord.Member,
+        until: Optional[datetime] = None,
+        reason: Optional[str] = None,
+    ) -> Dict[str, Optional[Union[discord.abc.GuildChannel, str, bool]]]:
+        """Mutes the specified user in the specified channel"""
+        overwrites = channel.overwrites_for(user)
+        permissions = channel.permissions_for(user)
+
+        if permissions.administrator:
+            return {
+                "success": False,
+                "channel": channel,
+                "reason": _(MUTE_UNMUTE_ISSUES["is_admin"]),
+            }
+
+        new_overs: dict = {}
+        move_channel = False
+        new_overs.update(send_messages=False, add_reactions=False, speak=False)
+        send_reason = None
+        if user.voice and user.voice.channel:
+            if channel.permissions_for(guild.me).move_members:
+                move_channel = True
+            else:
+                send_reason = _(MUTE_UNMUTE_ISSUES["voice_mute_permission"])
+
+        if not await self.is_allowed_by_hierarchy(guild, author, user):
+            return {
+                "success": False,
+                "channel": channel,
+                "reason": _(MUTE_UNMUTE_ISSUES["hierarchy_problem"]),
+            }
+
+        old_overs = {k: getattr(overwrites, k) for k in new_overs}
+        overwrites.update(**new_overs)
+        if channel.id not in self._channel_mutes:
+            self._channel_mutes[channel.id] = {}
+            self._channel_mutes[channel.id][user.id] = {
+                "author": author.id,
+                "member": user.id,
+                "until": until.timestamp() if until else None,
+            }
```

Comment:
Gotta unindent this for it to work 😬 
```suggestion

        self._channel_mutes[channel.id][user.id] = {
            "author": author.id,
            "member": user.id,
            "until": until.timestamp() if until else None,
        }
```